src/Aspire.Hosting.Testing/DistributedApplicationFactory.cs

```diff
@@ -336,7 +336,8 @@ private static void EnsureDepsFile(Type entryPoint)
         }
 
         var depsFileName = $"{entryPoint.Assembly.GetName().Name}.deps.json";
-        var depsFile = new FileInfo(Path.Combine(AppContext.BaseDirectory, depsFileName));
+        // entryPoint.Assembly.Location is always an absolute path
+        var depsFile = new FileInfo(Path.Combine(Path.GetDirectoryName(entryPoint.Assembly.Location)!, depsFileName));
         if (!depsFile.Exists)
         {
             throw new InvalidOperationException($"Missing deps file '{Path.GetFileName(depsFile.FullName)}'. Make sure the project has been built.");
```

Comment:
```suggestion
```

I don't think we need this code that checks if there is a deps.json file and we should delete it.

AFAICT, this was necessary way back in the original MVC days when we would explicitly read the .deps.json file. I've traced the code all the way back to https://github.com/aspnet/Mvc/pull/6708/files#diff-dfb443073875510f427cf8ae0eaccf7dac74bda2a3454892fbe0b3ad76e67b04L56, which DefaultAssemblyPartDiscoveryProvider would read the .deps.json file.

For .NET Aspire, I don't think this check adds value. And as can be seen in this PR, causes more issues.

Reply:
Dropped it.